components/script/dom/console.rs

```diff
@@ -65,13 +73,80 @@ fn console_message(global: &GlobalScope, message: DOMString, level: LogLevel) {
     })
 }
 
+//generate matrix of table elements
+#[allow(unsafe_code)]
+unsafe fn generate_table(
+    cx: &SafeJSContext,
+    tabular_data: &HandleValue,
+    _properties: &Option<Vec<DOMString>>,
+) -> Option<Vec<Vec<DOMString>>> {
+    let cx_ptr = *(cx.deref()) as *mut JSContext;
+
+    if !tabular_data.get().is_object() {
+        return None;
+    }
+
+    if is_array_like(cx_ptr, *tabular_data) {
+        let vec: Vec<DOMString> =
+            Vec::<DOMString>::from_jsval(cx_ptr, *tabular_data, StringificationBehavior::Empty)
+                .expect("Error when converting Array object to Vec<DOMString>")
+                .get_success_value()
+                .unwrap_or(&Vec::new())
+                .to_vec();
+
+        let mut res: Vec<Vec<DOMString>> = Vec::new();
+        res.push(vec!["(index)".into(), "Values".into()]);
+        for (index, word) in vec.into_iter().enumerate() {
+            res.push(vec![index.to_string().into(), word]);
+        }
```

Comment:
Does this implementation handle objects?

Reply:
The implementation doesn't handle objects. It just prints [object Object] when given an object, but I don't know how to retrieve the attributes of a JSObject and the documentation isn't helping much. Do you have any suggestions?